Updating react-table to v7 (#122)
OpenLiberty/draft-guide-rest-client-reactjs: README.adoc

----

ArtistTable.js
[source, javascript, linenums, role='code_column hide_tags=get-posts,useEffect,axios-library']
----
include::finish/src/main/frontend/src/Components/ArtistTable.js[]
----

> The [hotspot=react-library file=1]`React` library imports the `react` package for you to create the [hotspot=ArtistTable file=1]`ArtistTable` function needs to have the `export` declaration as it is being exported to the `App.js` module. The [hotspot=Posts file=1]`Posts` object is initialized using a React Hook which lets you add a state to represent the state of the posts that appear on the paginated table.

Reviewer: The `Posts`  object...
->
The `posts`  object...

Author: Done

> The [hotspot=react-library file=1]`React` library imports the `react` package for you to create the [hotspot=ArtistTable file=1]`ArtistTable` function needs to have the `export` declaration as it is being exported to the `App.js` module. The [hotspot=posts file=1]`posts` object is initialized using a React Hook which lets you add a state to represent the state of the posts that appear on the paginated table.

Reviewer 2: ```suggestion
The [hotspot=react-library file=1]`React` library imports the `react` package for you to create the [hotspot=ArtistTable file=1]`ArtistTable` function. THis function must have the `export` declaration because it is being exported to the `App.js` module. The [hotspot=posts file=1]`posts` object is initialized using a React Hook that lets you add a state to represent the state of the posts that appear on the paginated table.
```

Reviewer 2: I'm not sure exactly what the first sentence is trying to say- seems like something might be missing. My suggestion is my best guess but please review and adjust as needed.

To display the returned data, you will use pagination. Pagination is the process of separating content into discrete pages, and it can be used for handling data sets in React. In your application, you'll render the columns in the paginated table. The [hotspot=table-info file=1]`columns` constant is used to define the table that is present on the webpage.

> The [hotspot=useTable file=1]`useTable` hook is used to create a paginated table. The `useTable` hook takes in the [hotspot=table-info file=1]`columns` and [hotspot=Posts file=1]`Posts` objects as parameters. The `useTable` hook returns a paginated table that is assigned to the [hotspot=table file=1]`table` constant. The [hotspot=table file=1]`table` constant is used to render the paginated table on the webpage. The [hotspot=return-table file=1]`return` statement returns the paginated table. The [hotspot=useSortBy file=1]`useSortBy` hook is used to sort the paginated table by the column headers. The [hotspot=usePagination file=1]`usePagination` hook is used to create the pagination buttons that are used to navigate through the paginated table at the bottom of the table.

Reviewer: ... `Posts` objects ...
->
... `posts` and `setPosts` objects ...

Author: Done

> The [hotspot=useTable file=1]`useTable` hook is used to create a paginated table. The `useTable` hook takes in the [hotspot=table-info file=1]`columns` and [hotspot=posts file=1]`posts` and `setPosts` objects as parameters. The `useTable` hook returns a paginated table that is assigned to the [hotspot=table file=1]`table` constant. The [hotspot=table file=1]`table` constant is used to render the paginated table on the webpage. The [hotspot=return-table file=1]`return` statement returns the paginated table. The [hotspot=useSortBy file=1]`useSortBy` hook is used to sort the paginated table by the column headers. The [hotspot=usePagination file=1]`usePagination` hook is used to create the pagination buttons that are used to navigate through the paginated table at the bottom of the table.

Reviewer 2: ```suggestion
The [hotspot=useTable file=1]`useTable` hook creates a paginated table. The `useTable` hook takes in the [hotspot=table-info file=1]`columns`, [hotspot=posts file=1]`posts`, and `setPosts` objects as parameters. It returns a paginated table that is assigned to the [hotspot=table file=1]`table` constant. The [hotspot=table file=1]`table` constant renders the paginated table on the webpage. The [hotspot=return-table file=1]`return` statement returns the paginated table. The [hotspot=useSortBy file=1]`useSortBy` hook sorts the paginated table by the column headers. The [hotspot=usePagination file=1]`usePagination` hook creates the pagination buttons at the bottom of the table that are used to navigate through the paginated table.
```

----

[role="edit_command_text"]
Add the [hotspot=axios-library]`axios` library and the [hotspot=get-posts]`GetArtistsInfo()` function.

> The [hotspot=axios]`axios` HTTP call is used to read the artist JSON that contains the data from the sample JSON file in the `resources` directory. When a response is successful, the state of the system changes by assigning [hotspot=response-data]`response.data` to [hotspot=posts]`posts`. The [hotspot=for-artists]`artists` and their `albums` JSON data are manipulated to allow them to be accessed by the `ReactTable`. You will notice the [hotspot=spread-one]`...rest` or [hotspot=spread-two]`...album` object spread syntax is a relatively new sytnax made for simplicity. To learn more about it, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals[Spread in object literals^].

Reviewer 2: ```suggestion
The [hotspot=axios]`axios` HTTP call is used to read the artist JSON that contains the data from the sample JSON file in the `resources` directory. When a response is successful, the state of the system changes by assigning [hotspot=response-data]`response.data` to [hotspot=posts]`posts`. The [hotspot=for-artists]`artists` and their `albums` JSON data are manipulated to allow them to be accessed by the `ReactTable`. The [hotspot=spread-one]`...rest` or [hotspot=spread-two]`...album` object spread syntax is designed for simplicity. To learn more about it, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals[Spread in object literals^].
```

I've removed "relatively new" here only because that description is likely to go out of date over time.
